Sources/StreamChat/Workers/ChatClientUpdater.swift

```diff
@@ -11,110 +11,78 @@ class ChatClientUpdater {
+            completion(nil)
+
+        case .newToken:
+            client.updateWebSocketEndpoint(with: newToken, userInfo: userInfo)
+
+            guard newToken != client.currentToken else {
+                completion(nil)
                 return
             }
 
-            // Update the current user id to the new one.
-            client.currentUserId = newToken.userId
+            client.updateUser(with: newToken, completeTokenWaiters: true, isFirstConnection: false)
+            completion(nil)
+
+        case .newUser:
+            client.switchToNewUser(with: newToken)
 
-            // Update the current token with the new one.
-            client.currentToken = newToken
+            // Setting a new connection is not possible in connectionless mode.
+            guard client.config.isClientInActiveMode else {
+                completion(ClientError.ClientIsNotInActiveMode())
+                return
+            }
 
-            // Disconnect from web-socket.
             disconnect(source: .userInitiated) { [weak client] in
                 guard let client = client else {
                     completion(ClientError.ClientHasBeenDeallocated())
                     return
                 }
-                
-                // Update web-socket endpoint.
-                client.webSocketClient?.connectEndpoint = .webSocketConnect(
-                    userInfo: userInfo ?? .init(id: newToken.userId)
-                )
-
-                // Re-create backgroundWorker's since they are related to the previous user.
-                client.createBackgroundWorkers()
-                
-                // Stop tracking active components
-                client.activeChannelControllers.removeAllObjects()
-                client.activeChannelListControllers.removeAllObjects()
-                
-                // Reset all existing local data.
-                client.databaseContainer.removeAllData(force: true, completion: completion)
-            }
-            
-            return
-        }
-
-        // Set the web-socket endpoint
-        if client.webSocketClient?.connectEndpoint == nil {
-            client.webSocketClient?.connectEndpoint = .webSocketConnect(userInfo: userInfo ?? .init(id: newToken.userId))
-        }
 
-        guard newToken != client.currentToken else {
-            completion(nil)
-            return
-        }
-
-        client.currentToken = newToken
-
-        // Forward the new token to waiting requests.
-        client.completeTokenWaiters(token: newToken)
-
-        // It makes more sense to create background workers here
-        // other than in `init` because workers without currently logged-in
-        // user do nothing.
-        if client.backgroundWorkers.isEmpty {
-            client.createBackgroundWorkers()
+                client.updateWebSocketEndpoint(with: newToken, userInfo: userInfo)
+                client.clearPreviousUserData(completion: completion)
```

Comment:
Shouldn't these be reversed? Or it doesn't have any impact?

Reply:
No impact as of now, but will reverse it to seem a bit more logic